src/main/java/de/tum/in/www1/artemis/service/exam/ExamImportService.java

```diff
@@ -0,0 +1,322 @@
+    private final ProgrammingExerciseService programmingExerciseService;
+
+    private final ProgrammingExerciseRepository programmingExerciseRepository;
+
+    private final ProgrammingExerciseImportService programmingExerciseImportService;
+
+    private final FileUploadExerciseRepository fileUploadExerciseRepository;
+
+    private final FileUploadImportService fileUploadImportService;
+
+    private final GradingCriterionRepository gradingCriterionRepository;
+
+    public ExamImportService(TextExerciseImportService textExerciseImportService, TextExerciseRepository textExerciseRepository,
+            ModelingExerciseImportService modelingExerciseImportService, ModelingExerciseRepository modelingExerciseRepository, ExamRepository examRepository,
+            ExerciseGroupRepository exerciseGroupRepository, ExamAccessService examAccessService, QuizExerciseRepository quizExerciseRepository,
+            QuizExerciseImportService importQuizExercise, CourseRepository courseRepository, ProgrammingExerciseService programmingExerciseService1,
+            ProgrammingExerciseRepository programmingExerciseRepository, ProgrammingExerciseImportService programmingExerciseImportService,
+            FileUploadExerciseRepository fileUploadExerciseRepository, FileUploadImportService fileUploadImportService, GradingCriterionRepository gradingCriterionRepository) {
+        this.textExerciseImportService = textExerciseImportService;
+        this.textExerciseRepository = textExerciseRepository;
+        this.modelingExerciseImportService = modelingExerciseImportService;
+        this.modelingExerciseRepository = modelingExerciseRepository;
+        this.examRepository = examRepository;
+        this.exerciseGroupRepository = exerciseGroupRepository;
+        this.examAccessService = examAccessService;
+        this.quizExerciseRepository = quizExerciseRepository;
+        this.quizExerciseImportService = importQuizExercise;
+        this.courseRepository = courseRepository;
+        this.programmingExerciseService = programmingExerciseService1;
+        this.programmingExerciseRepository = programmingExerciseRepository;
+        this.programmingExerciseImportService = programmingExerciseImportService;
+        this.fileUploadExerciseRepository = fileUploadExerciseRepository;
+        this.fileUploadImportService = fileUploadImportService;
+        this.gradingCriterionRepository = gradingCriterionRepository;
+    }
+
+    /**
+     * Imports the given Exam with ExerciseGroups and Exercises to the given target Course
+     *
+     * @param examToCopy     the exam which should be copied together with exercise groups and exercises
+     * @param targetCourseId the course to which the exam should be imported
+     * @return the copied Exam with Exercise Groups and Exercises
+     */
+    @NotNull
+    public Exam importExamWithExercises(Exam examToCopy, long targetCourseId) {
+
+        examAccessService.checkCourseAccessForInstructorElseThrow(targetCourseId);
```

Comment:
This method is invoked via a REST endpoint, right? I would then consider moving this check into the corresponding Resource to not mix up the layering.